### DISCUSSION TITLE
Fetch live data using stlite or stlite desktop

### DISCUSSION THREAD

**Discussion by subhashsubramanian** [whitphx/stlite#611] [Q&A]
My streamlit app uses live data which is web-scraped using requests and beautiful soup. By deploying the streamlit app on the cloud, the website from where data is scraped may rate limit or block it as multiple users may use it at the same time. Can I address this with stlite? 

If I convert it to a stlite desktop exe, will it still be able to scrape the live data? 

Also, I don't want the Python code to be available to users if it is a desktop exe. 

Please advise me suitably

**Comment by whitphx** [comment]
I don't know about the API rate-limit, while at least `@stlite/desktop` itself doesn't do anything about networking stuff.

> Also, I don't want the Python code to be available to users if it is a desktop exe.

It's not possible by nature of `@stlite/desktop`. The discussion is #1048.
